holoviews/plotting/mpl/plot.py

```diff
@@ -82,8 +82,7 @@ class MPLPlot(DimensionedPlot):
     sublabel_size = param.Number(default=18, doc="""
          Size of optional subfigure label.""")
 
-    projection = param.ObjectSelector(default=None,
-                                      objects=['3d', 'polar', None], doc="""
+    projection = param.Parameter(default=None, doc="""
```

Comment:
To be honest creating the axes is one of the first things that happens and the traceback matplotlib provides is reasonable:

```python
ValueError [Call ipython.show_traceback() for details]
Unknown projection 'Test'
```